packages/block-library/src/cover/edit/poster-image.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * WordPress dependencies
+ */
+import { MediaUpload, MediaUploadCheck } from '@wordpress/block-editor';
+import {
+	Button,
+	BaseControl,
+	__experimentalHStack as HStack,
+	__experimentalToolsPanelItem as ToolsPanelItem,
+} from '@wordpress/components';
+import { __, sprintf } from '@wordpress/i18n';
+import { useRef } from '@wordpress/element';
+import { useInstanceId } from '@wordpress/compose';
+
+const COVER_POSTER_ALLOWED_MEDIA_TYPES = [ 'image' ];
+
+function PosterImage( { poster, setAttributes } ) {
+	const posterImageButton = useRef();
```

Comment:
```suggestion
	const posterButtonRef = useRef();
```

Suggestion: The new standard is to use `Ref` suffix to avoid React Compiler errors in the future. See #64718.